src/Command/DeleteFlagsHandler.php

```diff
@@ -53,7 +53,14 @@ public function handle(DeleteFlags $command)
 
         $this->events->dispatch(new FlagsWillBeDeleted($post, $actor, $command->data));
 
-        $post->flags()->delete();
+        //$post->flags()->delete();
+        $flags = $post->flags()->get();
+        foreach ($flags as $flag) {
+            $flag->dismissed_at = time();
+            $flag->dismissed_by_user_id = $actor->id;
+
+            $flag->save();
+        }
```

Comment:
Could we do this with a single query, rather than looping through the objects in PHP and creating a query for each of them?

Reply:
Yes, absolutely. Bad on my part for not making that more efficient.